test/Microsoft.AspNet.Mvc.TagHelpers.Test/LinkTagHelperTest.cs

```diff
@@ -0,0 +1,158 @@
+            // Arrange
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object>
+                {
+                    { "asp-fallback-href", "test.css" },
+                    { "asp-fallback-test-class", "hidden" },
+                    { "asp-fallback-test-property", "visible" },
+                    { "asp-fallback-test-value", "hidden" },
+                });
+            var output = MakeTagHelperOutput("link");
+            var loggerFactory = new Mock<ILoggerFactory>();
+
+            // Act
+            var helper = new LinkTagHelper
+            {
+                LoggerFactory = loggerFactory.Object,
+                FallbackHref = "test.css",
+                FallbackTestClass = "hidden",
+                FallbackTestProperty = "visible",
+                FallbackTestValue = "hidden"
+            };
+            helper.Process(context, output);
+
+            // Assert
+            Assert.Null(output.TagName);
+            Assert.NotNull(output.Content);
+            Assert.True(output.ContentSet);
+        }
+        
+        [Fact]
+        public void PreservesOrderOfSourceAttributesWhenRun()
+        {
+            // Arrange
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object>
+                {
+                    { "rel", "stylesheet"},
+                    { "data-extra", "something"},
+                    { "href", "test.css"},
+                    { "asp-fallback-href", "test.css" },
+                    { "asp-fallback-test-class", "hidden" },
+                    { "asp-fallback-test-property", "visible" },
+                    { "asp-fallback-test-value", "hidden" }
+                });
+            var output = MakeTagHelperOutput("link",
+                attributes: new Dictionary<string, string>
+                {
+                    { "rel", "stylesheet"},
+                    { "data-extra", "something"},
+                    { "href", "test.css"}
+                });
+            var loggerFactory = new Mock<ILoggerFactory>();
+
+            // Act
+            var helper = new LinkTagHelper
+            {
+                LoggerFactory = loggerFactory.Object,
+                FallbackHref = "test.css",
+                FallbackTestClass = "hidden",
+                FallbackTestProperty = "visible",
+                FallbackTestValue = "hidden"
+            };
+            helper.Process(context, output);
+
+            // Assert
+            Assert.StartsWith("<link rel=\"stylesheet\" data-extra=\"something\" href=\"test.css\"", output.Content);
+        }
+        
+        [Fact]
+        public void DoesNotRunWhenARequiredAttributeIsMissing()
+        {
+            // Arrange
+            var context = MakeTagHelperContext(
+                attributes: new Dictionary<string, object>
+                {
+                    // This is commented out on purpose: { "asp-fallback-href", "test.css" },
```

Comment:
instead of commenting out code, suggest the comment just read `// Left out "asp-fallback-href" to cause warning`